extensions/security/runtime-spi/src/main/java/io/quarkus/security/spi/runtime/AuthenticationEvent.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.security.spi.runtime;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.quarkus.security.identity.SecurityIdentity;
+
+public class AuthenticationEvent implements SecurityEvent {
```

Comment:
This event (as well as the authorization one), can it be used to indicate the authentication or authorization has succeeded ? IMHO it is should be supported too

Reply:
@michalvavrik As far as I know some deployments would like to have a journal of all type of security events, for example, to check afterwards how many times user A has logged in. On the OIDC side Security event of type OIDC LOGIN can help, but this is OIDC specific,

Your proposal sounds best, I'd only propose avoiding introducing hierarchy, IMHO, lets have `AuthenticationSuccessEvent`, `AuthenticationFailureEvent` and the same for the authorization, which will implement SecurityEvent  SPI and these 4 events will cover all the security event variations. Many users will indeed likely listen to authentication and authorization failures only.

By the way, in the next phase it can make sense provide some default implementation which will for ex log everything to Splunk or something like that